test/tests/functional/pbs_test_tpp.py

```diff
@@ -205,30 +269,40 @@ def common_steps(self, set_attr=None, exp_attr=None, job=False,
         :type resv_job: Bool. Defaults to False
         :param client: Name of the client
         :type client: String. Defaults to None
+        :param num_job: Number of jobs to be submitted
+        :type: Integer. Defaults to 1.
+        :param num_interactive: Number of interactive jobs to be submitted
+        :type num_interactive: Integer. Defaults to 1.
+        :param num_resv: Number of reservations to be submitted
+        :type num_resv: Integer. Defaults to 1.
         """
         if client is None:
             self.server.client = self.server.hostname
         else:
             self.server.client = client
-        if job:
-            jid = self.submit_job(
-                set_attr, exp_attr, job=True, job_script=True)
-            self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
-            self.server.log_match("%s;Exit_status=0" % jid)
-        if interactive:
-            # Submit Interactive Job
-            jid = self.submit_job(set_attr, exp_attr, interactive=True)
-            self.server.expect(JOB, 'queue', id=jid, op=UNSET)
-        if resv:
-            # Submit reservation
-            rid = self.submit_resv(resv_set_attr, resv_exp_attr)
-            jid = self.submit_job(set_attr, exp_attr, resv_job=True,
-                                  rid=rid, job_script=True)
-            # Wait for reservation to start
-            a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
-            self.server.expect(RESV, a, rid)
-            self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
-            self.server.log_match("%s;Exit_status=0" % jid)
+        if num_job > 1:
+            for i in (1, num_job+1):
+                if job:
+                    jid = self.submit_job(
+                        set_attr, exp_attr, job=True, job_script=True)
+                self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
+                self.server.log_match("%s;Exit_status=0" % jid)
+        if num_interactive > 0:
```

Comment:
This condition would always result true in case of default also. I suggest to check flag interactive first and then check num_interactive. Also, isn't the for loop missing in case of interactive and reservation submissions in case of num_interactive >1 and num_resv > 1?